WorldRestoration/src/main/java/as/minecraft/worldrestoration/tasks/PerformRegen.java

```diff
@@ -18,22 +24,56 @@
 	public void run() {
 		
 		//Test code to get and display coords of claim list
-		World world = Bukkit.getWorld("world");
-		ClaimHandler claimHandler = new ClaimHandler(world);
-		ArrayList<Vector3i[]> claimBounds = claimHandler.getWorldClaimBounds();
-		
-		for(Vector3i[] vec: claimBounds) {
-			Bukkit.broadcastMessage("Lesser Bound - X: " + Integer.toString(vec[0].getX()) + ", Z: " + Integer.toString(vec[0].getZ()));
-			Bukkit.broadcastMessage("Greater Bound - X: " + Integer.toString(vec[1].getX()) + ", Z: " + Integer.toString(vec[1].getZ()));
-			Bukkit.broadcastMessage("____________________________________");
+		Set<String> worldNames = DataStore.getWorldNames();
+		for(String worldName: worldNames) {
+			if(DataStore.getString("world-settings." + worldName + ".regen-enabled").trim().equals("true")) {
+				World world = Bukkit.getWorld(worldName);
+				int oldTime = Utils.getSecondsFromTimeUnit(DataStore.getString("world-settings." + worldName + ".considered-old-after").trim());
+				int restoreRadius = DataStore.getInt("world-settings." + worldName + ".radius");
+				ClaimHandler claimHandler = new ClaimHandler(world);
+				List<Rectangle> claimRectangles = claimHandler.getWorldClaimRectangles();
+				
+				Rectangle worldRectangle = new Rectangle(new Point(-restoreRadius, -restoreRadius), new Point(restoreRadius, restoreRadius));
+				List<Rectangle> rollbackRectangles = worldRectangle.subtractRectangles(claimRectangles);
+				
+				for(Rectangle rectangle: rollbackRectangles) {
+					List<Square> rollbackSquares = rectangle.convertToSquares();
+					for(Square square: rollbackSquares) {
+						
+						Location squareCentre = new Location(world, (double) (square.minPoint.x) + (double)(square.size)/2.0, 62.0, (double) (square.minPoint.y) + (double)(square.size)/2.0);
+						int squareRadius = (int) (Math.ceil((double)(square.size) / 2.0));
+						if(square.size % 2 == 0) squareRadius += 1;
+						
+						List<String[]> lookup = CoreProtectHook.coreProtect.performLookup(2147483600, null, null, null, null /*exclude-blocks*/, null /*actions*/, squareRadius, squareCentre);
+						if(checkListForOldEntry(oldTime, lookup)) {
+							performRestoration(oldTime, square, squareRadius, squareCentre);
+						}
+					}
+				}
+			}
 		}
-		//
-		
+	}
+	
+	private boolean checkListForOldEntry(int oldTime, List<String[]> lookup) {
+		if (lookup!=null && !lookup.isEmpty()) {
+			int currentTime = (int)(System.currentTimeMillis() / 1000L);
+			for(String[] entry: lookup) {
+				ParseResult result = CoreProtectHook.coreProtect.parseResult(entry);
+				//System.out.println("Time ago: " + (currentTime - result.getTime()) + ", rolled back: " + result.isRolledBack());
+				if(currentTime - result.getTime() > oldTime && !result.isRolledBack()) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+	
+	private void performRestoration(int oldTime, Square square, int radius, Location centre) {

```

Comment:
Ser ut som square er ubrukt.

Kanskje du kunne tatt inn bare `square`, og hatt funksjoner som `square.getOddRadius()` og `square.getCenter()`? Er ofte greit å holde antallet argumenter til det minimale, så forslaget mitt håndterer både ubrukt argument og antall argumenter.

Reply:
Mhm, jeg adda egentlig square.getCentre() i utgangspunktet, men fjerna det etterpå av et par grunner. Mulig dette kan sees på ved en senere anledning, for litt usikker på om jeg har lyst til å bruke Point getCentre() eller Location getCentre(). Bruker man Point må man konvertere til Location, men bruker man Location må man plutselig ha inn Bukkit.

Tror det er lurt å gjøre det ordentlig når man først fikser det, og eventuelt evaluere om man eventuelt bytter om Point.y til Point.z ettersom de horisontale aksene i Location og Minecraft er x og z med y oppover. For å gjøre det litt mindre forvirrende i konverteringer, som f.eks. "square.minPoint.y" for å finne minste z-koordinat i Location.

Reply:
Gode poeng. Vi kan se på det senere.